### PR TITLE
Add `tag_axes`

### PR DESCRIPTION
**Pull request by inducer**
Stolen from https://github.com/inducer/meshmode/pull/284.

### CODE REVIEW

File: arraycontext/context.py [hunk 1]
```diff
@@ -470,4 +477,23 @@ def permits_advanced_indexing(self) -> bool:
 
 # }}}
 
+
+# {{{ tagging helpers
+
+def tag_axes(
+        actx: ArrayContext,
+        dim_to_tags: Mapping[int, ToTagSetConvertible],
+        ary: ArrayT) -> ArrayT:
```
Comment (inducer) [inline arraycontext/context.py:486]: Note that I've changed the argument order here, compared to the original in https://github.com/inducer/meshmode/pull/284.

**Comment by inducer** [comment]
FYI I'm working on getting this merge-ready, so no need to mess with it ATM. :)

**Comment by inducer** [comment]
Unlike my earlier, misplace comment suggested, this should be ready for a look. Sorry about the confusion!

**Review by kaushikcfd — APPROVED** [review]
LGTM, thanks!

**Pull request merged**